### DISCUSSION TITLE
Help with social card not showing up

### DISCUSSION THREAD

**Discussion by johnsturgeon** [squidfunk/mkdocs-material#7332] [Q&A] [answered]
I just added the social plugin, and the preview cards are being rendered as I expect, but if I post a link to Mastodon, it doesn't render the card.

Here's an example post

https://mastodon.social/@johnsturgeon/112735617201938448

And here's the blog post:
https://johnsturgeon.me/2024/07/01/python-script-as-service/

Any help would be greatly appreciated.

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @johnsturgeon,
using for example these page inspectors:
- https://www.linkedin.com/post-inspector/inspect/https:%2F%2Fjohnsturgeon.me%2F2024%2F07%2F01%2Fpython-script-as-service%2F
- https://www.heymeta.com/url/johnsturgeon.me/2024/07/01/python-script-as-service/

we can see that the image doesn't load and the image URL points to:
- http://newblog.johnsturgeon.me/assets/images/social/posts/2024-07-01-python-script-as-service.png

taken from the website source:
```html
<meta  property="og:image"  content="https://newblog.johnsturgeon.me/assets/images/social/posts/2024-07-01-python-script-as-service.png" >
```

This link doesn't load at all. I also noticed that the base of the URL is different than your actual page, so I tried changing it:
- https://johnsturgeon.me/assets/images/social/posts/2024-07-01-python-script-as-service.png
and it works

So change your `site_url` in the `mkdocs.yml`:
- https://www.mkdocs.org/user-guide/configuration/#site_url
- https://github.com/johnsturgeon/blog/blob/161c53d80c4cea39d6a6e400076c9f9d929c223f/mkdocs.yml#L6

---

> [!NOTE]
>Also please keep in mind that services cache your shared URLs, so give it a couple of minutes / hours / re-shares to allow them to re-fetch your social cards.

**Comment by johnsturgeon** [comment]
You nailed it!  and Thank you for the links to test the cards, those are great resources!
